pkg/network/user_proxy.go

```diff
@@ -0,0 +1,56 @@
+package network
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/CESARBR/knot-babeltower/pkg/entities"
+	"github.com/CESARBR/knot-babeltower/pkg/logging"
+)
+
+// UserProxy proxy a request to the user service interface
+type UserProxy interface {
+	SendCreateUser(user entities.User) (err error)
+}
+
+// Proxy proxy a request to the user service
+type Proxy struct {
+	url    string
+	logger logging.Logger
+}
+
+// NewUserProxy creates a proxy to the users service
+func NewUserProxy(logger logging.Logger, hostname string, port uint16) *Proxy {
+	url := fmt.Sprintf("http://%s:%d", hostname, port)
+
+	logger.Debug("Proxy setup to " + url)
+	return &Proxy{url, logger}
+}
+
+// SendCreateUser proxy the http request to user service
+func (p *Proxy) SendCreateUser(user entities.User) (err error) {
+	var resp *http.Response
+
+	p.logger.Debug("Proxying request to create user")
+	client := &http.Client{Timeout: 10 * time.Second}
+	jsonUser, err := json.Marshal(user)
+	if err != nil {
+		goto done
+	}
+
+	resp, err = client.Post(p.url+"/users", "application/json", bytes.NewBuffer(jsonUser))
+	if err != nil {
+		goto done
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusConflict {
+		err = entities.ConflictResponse{}
```

Comment:
What do you think about preparing this error to receive a custom message so that we can explain the current context when throwing it? This message doesn't need to have the HTTP code because it's defined [here](https://github.com/CESARBR/knot-babeltower/pull/14/files#diff-c1671d516e41b74c29004a093569b58aR45) according to the error instance type.